src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -480,18 +460,8 @@ void CDSConfig::check_aot_flags() {
   CHECK_SINGLE_PATH(AOTCacheOutput);
   CHECK_SINGLE_PATH(AOTConfiguration);
 
-  if (FLAG_IS_DEFAULT(AOTCache) &&
-      FLAG_IS_DEFAULT(AOTMode)) {
-    bool has_cache_output = !FLAG_IS_DEFAULT(AOTCacheOutput);
-    bool has_config = !FLAG_IS_DEFAULT(AOTConfiguration);
-    if (!has_cache_output && !has_config) {
-      // AOT flags are not used. Use classic CDS workflow
-      return;
-    } else if (has_cache_output) {
-      // If AOTCacheOutput has been set, default mode is "record".
-      // Default value for AOTConfiguration, if necessary, will be assigned in check_aotmode_record().
-      FLAG_SET_ERGO(AOTMode, "record");
-    }
+  if (FLAG_IS_DEFAULT(AOTCache) && AOTAdapterCaching) {
+    log_debug(aot,codecache,init)("AOTCache is not specified - AOTAdapterCaching is ignored");
   }
   if (FLAG_IS_DEFAULT(AOTCache) && AOTStubCaching) {
     log_debug(aot,codecache,init)("AOTCache is not specified - AOTStubCaching is ignored");
```

Comment:
You can remove these checks. There is similar check in  `AOTCodeCache::initialize()` for all types of AOT code.

Reply:
Yes, I will do.